app/models/post.rb

```diff
@@ -0,0 +1,23 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_posts_counter
+
+  def update_posts_counter
+    update(comments_counter: comments.count)
+  end
```

Comment:
- [ ] [OPTIONAL] because you are using update(...) so it is more specific to a user, it should be kept in user model, also I am making it optional because you already have it in the user model, kindly remove it here